### ISSUE TITLE
Alert the user to what is new after updating NVDA

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#16961]
### Is your feature request related to a problem? Please describe.

In PR #16947, @XLTechie commented:

"wouldn't it make more sense to turn this on by default, let the new feature be exposed, and let those who don't like it figure out how to turn it off, instead of the other way around?

I too often run into the situation where I hear users say "I didn't know NVDA could even do that", just because we are too conservative and leave new features disabled by default. We assume that users will happen across them, recognize what they actually do, and recognize that what is done is something that they wanted all along."

(The specifics of that PR aren't important in this discussion, but I wanted to pull that comment out and flesh out the idea a bit).

@LeonarddeR went on to say:
"Thinking more about this, may be after every update, we should consider having a message that reminds one to open the what's new after an update?"

### Describe the solution you'd like

My suggestion is a dialog which pops up automatically the first time NVDA starts after an update with:
- The version number
- The text in between the version number and the "New features" heading in the changes.html file
- A link to the full what's new file
- A checkbox to "Show this dialog next time" which is unchecked by default.  If I am in a hurry now and need to do something, but want the prompt to come back to that info tomorrow I can check it and then dismiss the dialog.

A couple of caveats off the top of my head:
- If the user was updating to a point release of the current version, eg 2025.1 to 2025.1.1 - we would only want to show what is new in the point release, not everything new in that main version.
- If the "Welcome to NVDA" dialog is shown, I would not show the what's new dialog - if the user is new to NVDA then they already have enough dialogs to work through (welcome dialog, collection of statistics dialog etc), and may not be familiar with what is "old" in NVDA to appreciate what is new.

### Additional context

Happy to move this to a discussion to explore other options if anyone has a completely different approach in mind.  I thought the general idea was simple and defined enough to jump right to an issue.

**Comment by LeonarddeR** [comment]
Many thanks @Qchristensen for opening the issue. I think I can completely second your proposal.
Some additional thoughts:

1. We need to cover the case where updates from 2024.1 to 2025.1 (i.e. show everything that was between the previous and the current version). That suggests we need to store the last runned version somewhere in config, May be when running the update, the updater should record the current version before installing the new one. May be this is just too much and we should only care about the most recent version.
2. @Qchristensen Have you also considered making this an extension of the welcome dialog? I.e.
    a. Make `Show this dialog when NVDA starts` a tri-state combobox: yes, no and only after update
    a. Add the changes and the what's new button to the welcome dialog

**Comment by LeonarddeR** [comment]
Closing as duplicate of #14274.
